src/service/fixed-layer.js

```diff
@@ -617,19 +613,177 @@ export class FixedLayer {
 
     // Move element to the fixed layer.
     if (this.transfer_ && state.fixed && !oldFixed && state.transferrable) {
-      this.transferToTransferLayer_(fe, index, state);
+      this.getTransferLayer_().transferTo(fe, index, state);
     }
   }
 
   /**
-   * @param {!ElementDef} fe
-   * @param {number} index
-   * @param {!ElementStateDef} state
+   * @return {?TransferLayerDef}
+   */
+  getTransferLayer_() {
+    // This mode is only allowed for a single-doc case.
+    if (!this.transfer_ || this.transferLayer_) {
+      return this.transferLayer_;
+    }
+    const doc = this.ampdoc.win.document;
+    this.transferLayer_ =
+        doc.body.shadowRoot ?
+          new TransferLayerShadow(doc) :
+          new TransferLayerBody(doc);
+    return this.transferLayer_;
+  }
+
+  /**
+   * Find all `position:fixed` and `position:sticky` elements.
+   * @param {!Array<CSSRule>} rules
+   * @param {!Array<string>} foundSelectors
+   * @param {!Array<string>} stickySelectors
    * @private
    */
-  transferToTransferLayer_(fe, index, state) {
+  discoverSelectors_(rules, foundSelectors, stickySelectors) {
+    for (let i = 0; i < rules.length; i++) {
+      const rule = rules[i];
+      if (rule.type == /* CSSStyleRule */ 1) {
+        if (rule.selectorText != '*' && rule.style.position) {
+          if (rule.style.position == 'fixed') {
+            foundSelectors.push(rule.selectorText);
+          } else if (endsWith(rule.style.position, 'sticky')) {
+            stickySelectors.push(rule.selectorText);
+          }
+        }
+      } else if (rule.type == /* CSSMediaRule */ 4) {
+        this.discoverSelectors_(rule.cssRules, foundSelectors, stickySelectors);
+      } else if (rule.type == /* CSSSupportsRule */ 12) {
+        this.discoverSelectors_(rule.cssRules, foundSelectors, stickySelectors);
+      }
+    }
+  }
+}
+
+
+/**
+ * @typedef {{
+ *   id: string,
+ *   selectors: !Array,
+ *   element: !Element,
+ *   position: string,
+ *   placeholder: (?Element|undefined),
+ *   slot: (?Element|undefined),
+ *   fixedNow: boolean,
+ *   stickyNow: boolean,
+ *   top: (string|undefined),
+ *   transform: (string|undefined),
+ *   forceTransfer: (boolean|undefined),
+ * }}
+ */
+let ElementDef;
+
+
+/**
+ * @typedef {{
+ *   fixed: boolean,
+ *   sticky: boolean,
+ *   transferrable: boolean,
+ *   top: string,
+ *   zIndex: string,
+ * }}
+ */
+let ElementStateDef;
+
+
+/**
+ * The contract for transfer layer.
+ * @interface
+ */
+class TransferLayerDef {
```

Comment:
No, anything that's not directly referenced in JS syntax has to have "Def". Since this name is only referenced in closure's `@interface {...}`, eslint does not understand those comments and reports the name as unused. That's the only reason we are using "Def" suffix for - they are all whitelisted against eslint.